aten/src/ATen/templates/Tensor.h

```diff
@@ -199,7 +200,9 @@ class CAFFE2_API Tensor {
 
   DeprecatedTypeProperties & type() const {
     return globalDeprecatedTypePropertiesRegistry().getDeprecatedTypeProperties(
-        tensorTypeIdToBackend(type_id()), scalar_type());
+        tensorTypeIdToBackend(type_id()),
+        scalar_type(),
+        is_variable() && !at::NonVariableTypeMode::is_enabled());
```

Comment:
I'm not sure how to tell that you needed to check the variable type mode here. (As in, I could believe that this change is needed, but I don't know how to show it's correct.)

Reply:
I don't know if it's required, but I just want to align it with the way Type is picked for now.